presentation/src/main/java/com/yapp/bol/presentation/view/group/search/GroupSearchFragment.kt

```diff
@@ -0,0 +1,115 @@
+import com.yapp.bol.presentation.utils.loseFocusOnAction
+import com.yapp.bol.presentation.utils.textChangesToFlow
+import com.yapp.bol.presentation.utils.withLoadStateAdapters
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class GroupSearchFragment : BaseFragment<FragmentGroupSearchBinding>(R.layout.fragment_group_search) {
+
+    private val viewModel: GroupSearchViewModel by activityViewModels()
+
+    override fun onViewCreatedAction() {
+        super.onViewCreatedAction()
+        binding.lifecycleOwner = this@GroupSearchFragment
+        binding.viewGroupListLoading.root.visibility = View.GONE
+        bindView()
+    }
+
+    private fun bindView() {
+        val adapter = GroupListAdapter()
+        initPaging(adapter)
+        binding.initSearchView(adapter)
+        adapter.addOnPagesUpdatedListener {
+            binding.viewGroupListLoading.root.visibility = View.GONE
+        }
+    }
+
+    private fun initPaging(adapter: GroupListAdapter) {
+        val concatAdapter = adapter.withLoadStateAdapters(
+            header = GroupListLoadStateAdapter { adapter.retry() },
+            footer = GroupListLoadStateAdapter { adapter.retry() }
+        )
+        binding.rvGroupList.adapter = concatAdapter
+    }
+
+    // 상단 search view 초기화 관련 작업
+    private fun FragmentGroupSearchBinding.initSearchView(adapter: GroupListAdapter) {
+        binding.initEditText(adapter)
+
+        val editText = viewGroupSearch.etGroupSearch
+        val rightDrawableBtn = viewGroupSearch.btnSearchRight
+
+        rightDrawableBtn.setOnClickListener {
+            if (!editText.text.isNullOrEmpty() || !editText.text.isNullOrBlank()) {
+                editText.text.clear()
+            }
+        }
+
+        editText.loseFocusOnAction(EditorInfo.IME_ACTION_SEARCH, this.root.context)
+
+        binding.viewGroupSearch.btnCreateGroup.setOnClickListener {
+            // TODO : create transition
+        }
+    }
+
+    // search view의 edittext 세팅
+    private fun FragmentGroupSearchBinding.initEditText(adapter: GroupListAdapter) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            val editTextFlow = viewGroupSearch.etGroupSearch.textChangesToFlow()
+            val debounceDuration = 500L
+
+            editTextFlow
+                .onEach {
+                    val isTyping = !(it.isNullOrBlank() || it.isEmpty())
+                    viewGroupSearch.btnSearchRight.setImageButtonByState(isTyping)
+                }
+                .launchIn(this)
+
+            editTextFlow
+                .debounce(debounceDuration)
+                .onEach {
+                    adapter.searchByKeyword(it.toString())
+                }
+                .launchIn(this)
```

Comment:
p3 : editTextFlow 를 2번이나 collect 하는 이유가 뭔지 궁금해요.
```suggestion
             editTextFlow
                .onEach {
                    val isTyping = !(it.isNullOrBlank() || it.isEmpty())
                    viewGroupSearch.btnSearchRight.setImageButtonByState(isTyping)
                }
                .debounce(debounceDuration)
                .onEach {
                    adapter.searchByKeyword(it.toString())
                }
                .launchIn(this)
```
요렇게는 혹시 작성이 안될까요?

Reply:
요런 방법이..! 감사합니다~! 수정 완료~!!

[4bb53f3](https://github.com/YAPP-Github/22nd-Android-Team-2-Android/pull/54/commits/4bb53f3ac928fe464c1283bd57f2e7d9e72f2dfd)